src/app/icd/client/CheckInDelegate.h

```diff
@@ -39,6 +42,62 @@ class DLL_EXPORT CheckInDelegate
                                node that sent the check-in message.
      */
     virtual void OnCheckInComplete(const ICDClientInfo & clientInfo) = 0;
+
+    /**
+     * @brief Callback used to let the application know that a key refresh is
+     * needed to avoid counter rollover problems.
+     *
+     * The implementer of this function should dynamically create a new ICDRefreshKeyInfo object. They should also generate a new
+       key and store it in the ICDRefreshKeyInfo object.
+       The implementer of this function should use CreateICDRefreshKeyInfo() to create the object
```

Comment:
Why does CreateICDRefreshKeyInfo even exist as API on the delegate?  Isn't the the SDK will call this OnKeyRefreshNeeded API?

This seems to be putting implementation details (CreateICDRefreshKeyInfo) into API for some reason...

Reply:
Removed it from CheckInDelegate.